### PR TITLE
Add celery task for migrating subscriptions

### PR DESCRIPTION
**Pull request by rudigiesler**
This PR aims to put in place the framework for running the celery task resposible for migrating the subscriptions. This involves fetching and cycling through the identities, updating the MigrateSubscription objects so that the latest status is viewable on the front end, as well as logging and displaying those logs on the front end.

This PR is not concerned with the actual migration of subscriptions, that will be implemented in a later stage.

### CODE REVIEW

File: mapper/tests.py [hunk 1]
```diff
@@ -344,10 +352,269 @@ def test_form_correct_submission(self):
...
+            raise Exception('Test error')
+        count_identities.side_effect = error_effect
+
+        migrate_subscriptions.delay(migrate.pk)
+
+        log_args = log.call_args[0]
+        self.assertEqual(log_args[0], migrate)
+        self.assertEqual(log_args[1], logging.ERROR)
+        self.assertTrue('Exception' in log_args[2])
+        self.assertTrue('Test error' in log_args[2])
```
Comment (KaitCrawford) [inline mapper/tests.py:582]: Shouldn't we check the MigrateSubscription objects status field was updated as well?

File: mapper/tests.py [hunk 2]
```diff
@@ -344,10 +352,269 @@ def test_form_correct_submission(self):
...
+        self.assertNotEqual(migrate.task_id, None)
+        self.assertNotEqual(migrate.completed_at, None)
+        self.assertEqual(migrate.total, 2)
+        self.assertEqual(migrate.current, 2)
+
+    @mock.patch('mapper.tasks.MigrateSubscriptionsTask.log')
+    @mock.patch('mapper.tasks.MigrateSubscriptionsTask.count_identities')
+    def test_run_failure_args(self, count_identities, log):
+        """
+        If the task raises an exception, and the migrateion object was provided
```
Comment (KaitCrawford) [inline mapper/tests.py]: small spelling error `migration`

File: mapper/tests.py [hunk 3]
```diff
@@ -344,10 +352,269 @@ def test_form_correct_submission(self):
...
+        self.assertEqual(log_args[1], logging.ERROR)
+        self.assertTrue('Exception' in log_args[2])
+        self.assertTrue('Test error' in log_args[2])
+
+    @mock.patch('mapper.tasks.MigrateSubscriptionsTask.log')
+    @mock.patch('mapper.tasks.MigrateSubscriptionsTask.count_identities')
+    def test_run_failure_kwargs(self, count_identities, log):
         """
-        Ensures that the test task can run.
+        If the task raises an exception, and the migrateion object was provided
```
Comment (KaitCrawford) [inline mapper/tests.py]: small spelling error `migration`

File: mapper/tests.py [hunk 4]
```diff
@@ -344,10 +352,269 @@ def test_form_correct_submission(self):
...
+            raise Exception('Test error')
+        count_identities.side_effect = error_effect
+
+        migrate_subscriptions.delay(migrate_subscription_id=migrate.pk)
+
+        log_args = log.call_args[0]
+        self.assertEqual(log_args[0], migrate)
+        self.assertEqual(log_args[1], logging.ERROR)
+        self.assertTrue('Exception' in log_args[2])
+        self.assertTrue('Test error' in log_args[2])
```
Comment (KaitCrawford) [inline mapper/tests.py:609]: Same here about checking the status

**Review by KaitCrawford — COMMENTED** [review]
Just two small changes to the tests and two tiny spelling mistakes

**Comment by rudigiesler** [comment]
@KaitCrawford Thanks for the review. I've addressed your comments, ready for another review.

**Review by KaitCrawford — APPROVED** [review]
:+1:

**Pull request merged**